File: Lab5/Mobile/app/src/main/java/com/example/mycharityapp/model/Donation.kt
package com.example.mycharityapp.model

import com.google.gson.annotations.SerializedName

data class Donation(
    @SerializedName("_id")
    val id: String,
    val title: String,
    val goal: Int,
    val description: String,
    val raised: Int,
    val timestamp: Long,
    val creatorId: String,
    val news: List<NewsItem>
)


data class NewsItem(
    val id: String,
    val date: String,
    val content: String
)

File: Lab5/Backend/models/Donation.js
const mongoose = require('mongoose')

const newsSchema = new mongoose.Schema({
  id: String,
  date: String,
  content: String
})

const donationSchema = new mongoose.Schema({
  _id: String, 
  title: String,
  goal: Number,
  description: String,
  raised: {
    type: Number,
    default: 0
  },
  timestamp: {
    type: Number,
    default: Date.now
  },
  creatorId: String,
  news: [newsSchema]
})



module.exports = mongoose.model('Donation', donationSchema)
